### PR TITLE
feat(actions-global): to add use-on-action-error

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

Similar to the prior PR this adds useOnActionError.

### CODE REVIEW

**Comment by tdeekens** [comment]
The `handleActionError` return seems to be https://github.com/commercetools/merchant-center-application-kit/blob/master/packages/actions-global/src/actions/handle-action-error.ts#L31 but  this indicates it can return null (https://github.com/commercetools/merchant-center-application-kit/blob/master/packages/actions-global/src/actions/handle-action-error.ts#L51). Maybe that's why the useActionError doesn't pass the type check.

File: packages/actions-global/src/hooks/use-on-action-error.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { Dispatch } from 'redux';
+import { useDispatch } from 'react-redux';
+import { handleActionError } from '../actions';
+
+export default function useOnActionError(error) {
+  const dispatch = useDispatch<
+    Dispatch<ReturnType<typeof handleActionError>>
+  >();
+  return React.useCallback(() => {
+    dispatch(handleActionError(error));
```
Comment (emmenko) [inline packages/actions-global/src/hooks/use-on-action-error.ts]: You need to separate this call.

```suggestion
    const action = handleActionError(error);
    if (action) dispatch(action);
```

Comment (tdeekens) [reply packages/actions-global/src/hooks/use-on-action-error.ts]: Thanks TypeScript. Thanks @emmenko :)

File: packages/actions-global/src/hooks/use-on-action-error.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { useDispatch } from 'react-redux';
+import { handleActionError } from '../actions';
+
+export default function useOnActionError(error) {
+  const dispatch = useDispatch<
+    Dispatch<ReturnType<typeof handleActionError>>
+  >();
+  return React.useCallback(() => {
+    const action = handleActionError(error);
+    if (action) dispatch(action);
```
Comment (emmenko) [inline packages/actions-global/src/hooks/use-on-action-error.ts]:
```suggestion
    if (action) return dispatch(action);
```

We should probably return this

**Comment by emmenko** [comment]
Ah wait, let me have a look

File: packages/actions-global/src/hooks/use-on-action-error.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import handleActionError, {
+  ActionError,
+  DispatchActionError,
+} from '../actions/handle-action-error';
+
+export default function useOnActionError(error: ActionError) {
+  const dispatch = useDispatch<DispatchActionError>();
+  return React.useCallback(() => handleActionError(error)(dispatch), [
```
Comment (emmenko) [inline packages/actions-global/src/hooks/use-on-action-error.ts]: @tdeekens note that it's not `dispatch(handleActionError(error))` but `handleActionError` is a factory function

**Pull request merged**